dataplane/linux/int_dataplane.go

```diff
@@ -467,7 +467,7 @@ func NewIntDataplaneDriver(config Config) *InternalDataplane {
 	}
 
 	// TODO Integrate XDP and BPF infra.
-	if !config.BPFEnabled && dp.xdpState == nil {
+	if !config.BPFEnabled && config.XDPEnabled && dp.xdpState == nil {
```

Comment:
We would like this code to run, even when `config.XDPEnabled` is false, so that Felix can clean up its own XDP state after a restart.

Reply:
Okay, I'll revert this pull request.